rustls/src/client/client_conn.rs

```diff
@@ -635,6 +647,12 @@ impl ClientConnection {
         self.inner.core.is_early_data_accepted()
     }
 
+    /// Access connection options whose use is dangerous and requires
+    /// extra care.
+    pub fn dangerous(self) -> danger::DangerousClientConnection {
```

Comment:
I think this wants to be called `into_dangerous` to reflect that it consumes the connection. That also means it isn't a more general venue for other dangerous APIs that don't require consuming the connection.

Reply:
> That also means it isn't a more general venue for other dangerous APIs that don't require consuming the connection.

Good point, maybe that's an argument in favour of [djc's suggestion](https://github.com/rustls/rustls/pull/1484#discussion_r1332919178) to use an extension trait instead. WDYT about that proposal?

Reply:
> Can we just do dangerous_extract_secrets(self)?

:+1:, implemented in-branch. I think with the separate `enable_secret_extraction` bool gating this from being useful until you've taken action to enable it having the name imply the danger seems sufficient.